app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -1,6 +1,15 @@
 class OmniauthCallbacksController < Devise::OmniauthCallbacksController
   include Devise::Controllers::Rememberable
 
+  # Rails actionpack only allows POST requests that come with an ORIGIN header
+  # that matches `request.base_url`, it raises CSRF exception otherwise.
+  # There is no way to allow specific ORIGIN values in order to securely bypass
+  # trusted origins (i.e. Apple OAuth) so `protect_from_forgery` is skipped
+  # ONLY when it's safe to do so (i.e. ORIGIN == 'https://appleid.apple.com').
+  # The hardcoded CSRF check can be found in the method `valid_request_origin?`:
+  # https://github.com/rails/rails/blob/master/actionpack/lib/action_controller/metal/request_forgery_protection.rb#L453-L463
+  protect_from_forgery unless: -> { safe_apple_callback_request? }
```

Comment:
I'm okay with having this auth callback bypass `protect_from_forgery ` because SIWA uses a signed JWT to pass in the Auth data to the callback:

> the JWT travels through the HTTPS request so I think I'm comfortable with that

[Extracted from conversation about this here](https://dev.to/fdoxyz/comment/14jdl)

I'm also happy about the approach taken to bypass this, because we're only doing so **if it's coming from Apple's trusted path**. An analogy I've used for this before is:

> Instead of leaving this door unlocked for anyone we are only giving 1 trusted path (`https://appleid.apple.com`) the keys to pass through the door